src/accelerate/utils/bnb.py

```diff
@@ -334,25 +334,26 @@ def _replace_with_bnb_layers(
                     break
             if proceed:
                 # Load bnb module with empty weight and replace ``nn.Linear` module
-                if bnb_quantization_config.load_in_8bit:
-                    bnb_module = bnb.nn.Linear8bitLt(
-                        module.in_features,
-                        module.out_features,
-                        module.bias is not None,
-                        has_fp16_weights=False,
-                        threshold=bnb_quantization_config.llm_int8_threshold,
-                    )
-                elif bnb_quantization_config.load_in_4bit:
-                    bnb_module = bnb.nn.Linear4bit(
-                        module.in_features,
-                        module.out_features,
-                        module.bias is not None,
-                        bnb_quantization_config.bnb_4bit_compute_dtype,
-                        compress_statistics=bnb_quantization_config.bnb_4bit_use_double_quant,
-                        quant_type=bnb_quantization_config.bnb_4bit_quant_type,
-                    )
-                else:
-                    raise ValueError("load_in_8bit and load_in_4bit can't be both False")
+                with torch.device(module.weight.device): # in case of meta device
```

Comment:
Why did you add this ? This function already under `init_empty_weights` in the `load_and_quantize_model` function

Reply:
In docs https://huggingface.co/docs/accelerate/usage_guides/quantization, it was not clear. Plus `load_and_quantize_model` gives a simple warning in case model_device.type != "meta"`` . So I thought `load_and_quantize_model` works for all types of devices.

Reply:
Oh thanks, if you initialized the model with `init_empty_weights`, `model_device.type` should be "meta" no ? Could you check why it is not returning meta ? Maybe we need to change the `model_device.type` check instead

Reply:
I run my script from accelerate cli to perform deepspeed stage 2 on cluster of machines. Do not know if this has an effect on code.

Reply:
Not sure also. Feel free to open a PR if you manage to fix the issue ! We can discuss about that in the issue you create. If you can share a minimal reproducer , that would be helpful. The best would be to reproduce this without deepspeed.